### DISCUSSION TITLE
Remove the NVDA menu on secure screens

### DISCUSSION THREAD

**Discussion by trypsynth** [nvaccess/nvda#13844] [Ideas]
We've seen a lot of security problems from having the NVDA menu accessible from secure screens, and honestly I se no reason for this. I don't see why we can change settings when you are unable to save your config, or pull up an about dialog, or use the speech/braille viewers. Secure screens are something you come by in passing (e.g. in a user account control dialog, or when logging in), not something you spend a lot of time in. It does have its uses, for example with things like KillNVDA, but I don't see any reason that can't be bound to a hotkey, not in the NVDA menu.

Thoughts?

**Comment by seanbudd** [comment]
When using a shared computer, a user may need to access certain settings (e.g. selecting a braille driver, or synthesizer voice).

It would be good to figure out the minimum set of settings to make a simpler menu and settings dialog for secure screens.

**Comment by trypsynth** [comment]
@seanbudd, possibly, but keep in mind how little time you typically spend on a secure screen. When a UAC dialog pops up, alt+y, when the sign in screen, you're focused on the password field right away, etc etc. There is something to be said for being able to adjust settings on a secure screen, but is it honestly worth all that, considering how little time you spend there, as well as the fact that you won't be able to save that?

**Comment by feerrenrut** [comment]
@TheQuinbox the use-case @seanbudd is alluding to is a kiosk or shared computer (EG a school library computer). An example of this need is a user who is connecting their specific braille display and needs to manually configure NVDA before they can log-in.

With HID braille and auto detection, this configuration may no longer be required. Other configuration would need to be assessed.

NVDA errs on the side of exposing more settings to ensure no one is locked out by design. This does leave NVDA open to potentially inconvenient changes by unauthenticated users. Note, to mitigate this you can always:
- reload the saved config.
- restart NVDA.
- or reset NVDA to factory default settings.

Access to settings on secure screens could be made configurable. Advanced users could disable access to settings on secure screens, preventing settings from being changed even temporarily. This would require:
- block access to the NVDA menu, settings dialogs, symbols, dictionaries etc
- prevent usage of synth settings ring.
- find and prevent usage of commands / scripts that affect settings.
